wurst/_items/OraclePotion.wurst

```diff
@@ -0,0 +1,120 @@
+    ..setTooltipNormalExtended(1, "This unit can see invisible unit.")
+    ..setIcon("ReplaceableTextures\\PassiveButtons\\PASBTNMagicalSentry.blp")
+ 
+@compiletime function createDummyAbility()
+    new AbilityDefinitionWindWalk(ABILITY_DUMMY_ID)
+    ..setArtCaster("")
+    ..setArtEffect("")
+    ..setArtTarget("")
+    ..setArtSpecial("")
+    ..setIconNormal(SPELL_ICON)
+    ..setIconResearch(SPELL_ICON)
+    ..setIconTurnOff(SPELL_ICON)
+    ..setHeroAbility(false)
+    ..setItemAbility(true)
+    ..setLevels(1)
+    ..setHotkeyNormal("A")
+    ..setName("TrueSight")
+    ..setTooltipNormal(1, "dummy spell")
+    ..setTooltipNormalExtended(1, "dummy spell")
+    ..setCooldown(1, COOLDOWN)
+    ..setManaCost(1, 0)
+    ..setAnimationNames("")
+    ..setDurationHero(1, 0.01)
+    ..setCastingTime(1, 0)
+    ..setDurationNormal(1, 0.01)
+    ..setEffectSound("")
+
+@compiletime function createDummyAura()
+    new AbilityDefinitionAuraSlow(BUFF_AURA_ID)
+    ..setArtTarget("Abilities\\Spells\\Human\\MagicSentry\\MagicSentryCaster.mdl")
+    ..setTargetAttachmentPoint("overhead")
+    ..setMovementSpeedFactor(1, 0)
+    ..setAreaofEffect(1, 0)
+    ..setBuffs(1, toRawCode(BUFF_ID))
+    ..setDurationHero(1, 20)
+    ..setDurationNormal(1, 20)
+    ..setTargetsAllowed(1, "self")
+    ..setName("True Sight")
+ 
+@compiletime function createTrueSightSpell()
+    new AbilityDefinition(ABILITY_ID, 'Atru')
+    ..setCastRange(1, 900)
+    ..setName("Oracle clarity")
+    ..setTooltipNormal(1, "True Sight")
+    ..setTooltipNormalExtended(1, "Reveal invisible unit")
+    ..setBuffs(1, "Binf")
+    ..setItemAbility(true)
+    ..setDurationHero(1, 20)
+    ..setDurationNormal(1, 20)
+    ..setButtonPositionNormalY(0)
+    ..setArtCaster("Abilities\\Spells\\Human\\MagicSentry\\MagicSentryCaster.mdl")
+    ..setCasterAttachmentPoint("overhead")
+    ..setCasterAttachments(0)
+
+@compiletime function createOraclePotion()
+    new ItemDefinition(ITEM_ORACLE_POTION, 'pinv')
+    ..setName("Oracle Potion")
+    ..setAbilities(toRawCode(ABILITY_DUMMY_ID))
+    ..setActivelyUsed(true)
+    ..setTooltipExtended("The consumer of this potion gain the true sight buff for 20 seconds, allowing him to see stealthed unit. Good versus thief & vision ward")
+    ..setModelUsed("Abilities\\Spells\\Other\\AcidBomb\\BottleMissile.mdx")
+    ..setLumberCost(11)
+    ..setGoldCost(0)
+    ..setInterfaceIcon("ReplaceableTextures\\CommandButtons\\BTNCharm.blp")
+    ..setDescription("Give true sight")
+    ..setDroppedWhenCarrierDies(true)
+    ..setNumberofCharges(1)
+
+HashMap<unit, timer> removalTimers = new HashMap<unit,timer>()
+
+function onCast()
+    var caster = GetSpellAbilityUnit()
+
+    if removalTimers.has(caster)
+        caster.removeAbility(SPELLBOOK_ABILITY_ID)
+        caster.removeAbility(BUFF_AURA_ID)
+        removalTimers.getAndRemove(caster).release()
+    caster.addAbility(BUFF_AURA_ID)
+    caster.addAbility(SPELLBOOK_ABILITY_ID)
```

Comment:
I think adding the spellbook is enough, since the aura is inside the book anyway